src/Api/Models/Response/PlanResponseModel.cs

```diff
@@ -36,9 +36,50 @@ public PlanResponseModel(Plan plan, string obj = "plan")
         DisplaySortOrder = plan.DisplaySortOrder;
         LegacyYear = plan.LegacyYear;
         Disabled = plan.Disabled;
-        SupportsSecretsManager = plan.SupportsSecretsManager;
-        SecretsManager = plan.SecretsManager;
-        PasswordManager = plan.PasswordManager;
+        if (plan.SecretsManager != null)
+        {
+            SecretsManager = new SecretsManagerPlanResponseModel
+            {
+                MaxServiceAccounts = plan.SecretsManager.MaxServiceAccounts,
+                AllowServiceAccountsAutoscale = plan.SecretsManager is { AllowServiceAccountsAutoscale: true },
+                StripeServiceAccountPlanId = plan.SecretsManager.StripeServiceAccountPlanId,
+                AdditionalPricePerServiceAccount = plan.SecretsManager.AdditionalPricePerServiceAccount,
+                BaseServiceAccount = plan.SecretsManager.BaseServiceAccount,
+                MaxAdditionalServiceAccount = plan.SecretsManager.MaxAdditionalServiceAccount,
+                HasAdditionalServiceAccountOption = plan.SecretsManager is { HasAdditionalServiceAccountOption: true },
+                StripeSeatPlanId = plan.SecretsManager.StripeSeatPlanId,
+                HasAdditionalSeatsOption = plan.SecretsManager is { HasAdditionalSeatsOption: true },
+                BasePrice = plan.SecretsManager.BasePrice,
+                SeatPrice = plan.SecretsManager.SeatPrice,
+                BaseSeats = plan.SecretsManager.BaseSeats,
+                MaxSeats = plan.SecretsManager.MaxSeats,
+                MaxAdditionalSeats = plan.SecretsManager.MaxAdditionalSeats,
+                AllowSeatAutoscale = plan.SecretsManager.AllowSeatAutoscale,
+                MaxProjects = plan.SecretsManager.MaxProjects
+            };
```

Comment:
Move this mapping logic into the constructor, so that it takes a `SecretsManagerPlanFeatures` object.

e.g. 

```cs
       if (plan.SecretsManager != null)
        {
            SecretsManager = new SecretsManagerPlanResponseModel(plan.SecretsManager)
        }
```

Reply:
Also, at the risk of being too long, I think this should be called `SecretsManagerPlanFeaturesResponseModel` for consistency.